core/src/main/java/org/elasticsearch/common/io/PathUtils.java

```diff
@@ -57,10 +57,30 @@ private PathUtils() {}
      * a path against an existing one!
      */
     public static Path get(String first, String... more) {
-        return DEFAULT.getPath(first, more);
+        return get(DEFAULT, first, more);
+    }
+
+    /**
+     * Returns a {@code Path} from name components against the given
+     * {@code FileSystem}.
+     * <p>
+     * This works just like {@code Paths.get()}.
+     * Remember: just like {@code Paths.get()} this is NOT A STRING CONCATENATION
+     * UTILITY FUNCTION.
+     * <p>
+     * Remember: this should almost never be used. Usually resolve
+     * a path against an existing one!
+     *
+     * @param fs    the given {@code FileSystem}
+     * @param first the first path component
+     * @param more  the remaining path components
+     * @return a path
+     */
+    public static Path get(FileSystem fs, String first, String... more) {
+        return fs.getPath(first, more);
```

Comment:
I don't understand the utility of this added method. Callers could just call it directly on the filesystem they already have?